R/SupervisedTask.R

```diff
@@ -1,13 +1,13 @@
-makeSupervisedTask = function(type, data, target, weights, blocking, fixup.data, check.data) {
-  task = makeTask(type = type, data = data, weights = weights, blocking = blocking, fixup.data = fixup.data, check.data = check.data)
+makeSupervisedTask = function(type, data, target, weights, blocking, fixup.data, check.data, spatial) {
+  task = makeTask(type = type, data = data, weights = weights, blocking = blocking, fixup.data = fixup.data, check.data = check.data, spatial = spatial)
 
   if (check.data) {
     # costsens does not have a target col...
     # assertCharacter(target, any.missing = FALSE, min.len = 1L)
     w = which.first(target %nin% colnames(data))
     if (length(w) > 0L)
       stopf("Column names of data doesn't contain target var: %s", target[w])
-    checkTaskData(task$env$data, cols = setdiff(colnames(data), target))
+    checkTaskData(task$env$data, cols = setdiff(colnames(task$env$data), target))
```

Comment:
I thought about this. One of my first implementations indeed stored `x` and `y` within `task$task.desc` instead of `data`. 

One problem that occurred was that I need `x` and `y` to follow along the dimensions of `data` so that it always fits the data and partitioning (especially in a nested case) is performed correctly. 

Keeping `x` and `y` at `data` has the opportunity that it gets automatically subsetted with `data` and only needs to be removed once (before the training + predict step).

Placing `x` and `y` in `task$task.desc` would mean that I would need to do an additional subsetting step every time something changed in `data` to keep `x` and `y` up to date with `data`.  This would happen way more often than the removal of `x` and `y` before "train + predict". Therefore I think it is more error-prone and would affect runtime more than the other approach.